scripts/reset-simulators.rb

```diff
@@ -17,55 +20,70 @@ def platform_for_device_type(device_type)
+      sleep 0.2 # we are sending AppleEvents, not a kill signal
+    end
+    
+    # kill any instances of simctl, since they might hold onto (or restart) the CoreSimulatorService service
+    `/usr/bin/pkill -qx simctl`
+    
+    # stop CoreSimulatorService
+    `/bin/launchctl remove com.apple.CoreSimulator.CoreSimulatorService 2>/dev/null`
+    sleep 0.2 # launchtl can take a moment to kill services
+    
+    # kill anything remaining with fire
+    until `/usr/bin/pgrep -x Simulator 'Simulator (Watch)' simctl com.apple.CoreSimulator.CoreSimulatorService`.empty?
+      `/usr/bin/pkill -9 -x Simulator 'Simulator (Watch)' simctl com.apple.CoreSimulator.CoreSimulatorService`
+      sleep 0.2
+    end
 
-  # Kill all the current simulator processes as they may be from a different Xcode version
-  print 'Killing running Simulator processes...'
-  while system('pgrep -q Simulator')
-    system('pkill Simulator 2>/dev/null')
-    # CoreSimulatorService doesn't exit when sent SIGTERM
-    system('pkill -9 Simulator 2>/dev/null')
+    puts ' done!'
+  end
+  
+  # - Switching simulator versions often causes CoreSimulatorService to throw an exception.
+  print 'Waiting for simulator to be ready...'
+  while `xcrun simctl list devices 2>/dev/null`.empty?
+    sleep 0.2
   end
   puts ' done!'
 
-  wait_for_core_simulator_service
-
-  # Shut down any running simulator devices. This may take multiple attempts if some
-  # simulators are currently in the process of booting or being created.
+  # - Shut down any running simulator devices.
+  # This may take multiple attempts if some simulators are in-flight
   all_available_devices = []
   (0..5).each do |shutdown_attempt|
     devices_json = `xcrun simctl list devices -j`
-    all_devices = JSON.parse(devices_json)['devices'].flat_map { |_, devices| devices }
-
-    # Exclude devices marked as unavailable as they're from a different version of Xcode.
-    all_available_devices = all_devices.reject { |device| device['availability'] =~ /unavailable/ }
-
-    break if running_devices(all_available_devices).empty?
-
-    shutdown_simulator_devices all_available_devices
+    raise 'xcrun failed!' unless $?.success?
+    all_available_devices = JSON.parse(devices_json)['devices'].flat_map { |_, devices| devices }
+
+    # Include only available devices, others belong to other versions of the simulator
+    all_available_devices = all_available_devices.select { |device| device['availability'] == '(available)' }
+    
+    # Get the list of non-shutdown devices
+    running_devices = all_available_devices.reject { |device| device['state'] == 'Shutdown' }
+    
+    break if running_devices.empty?
+    
+    # Shutdown the running devices
+    running_devices.each do |device|
+      puts "  Shutting down simulator #{device['udid']}"
+      system("xcrun simctl shutdown #{device['udid']}") or puts "    Failed to shut down simulator #{device['udid']}"
+    end
```

Comment:
I'm not really clear why you've chosen to move code that was previously in functions and inline it here. It makes it harder to see what's changed.